src/main/java/com/zaxxer/nuprocess/windows/ProcessCompletions.java

```diff
@@ -237,7 +237,7 @@ private void queueWrite(final WindowsProcess process)
 
    private void queueRead(final WindowsProcess process, final WindowsProcess.PipeBundle pipe, final int stdX)
    {
-      if (NuKernel32.ReadFile(pipe.pipeHandle, pipe.bufferPointer, NuProcess.BUFFER_CAPACITY, null, pipe.overlapped) == 0) {
+      if (NuKernel32.ReadFile(pipe.pipeHandle, pipe.bufferPointer.share(pipe.buffer.position()), pipe.buffer.remaining(), null, pipe.overlapped) == 0) {
```

Comment:
I think it'll be handled by the above check I added after `onStdout()` / `onStderr()` return, but I added another check here and we now throw `RuntimeException` in this case.